htdocs/compta/facture/class/api_invoices.class.php

```diff
@@ -666,7 +666,7 @@ public function delete($id)
 		}
 
 		$result = $this->invoice->delete(DolibarrApiAccess::$user);
-		if ($result < 0) {
+		if ($result <= 0) {
```

Comment:
The case 0 is not a technical error so it must not return the Exception with code 500
Add a if to return a
throw new RestException(403, 'Invoice not erasable');